code/game/gamemodes/distress.dm

```diff
@@ -38,11 +38,11 @@
 
 /datum/game_mode/distress/pre_setup()
 	. = ..()
-
+	var/number_of_xenos = length(xenomorphs)
```

Comment:
this should be in hivedatum code not here

Reply:
This happens before any xenos have spawned, the hive is empty.
It's a measure of how many players have been successfully chosen to spawn as xenos at roundstart, no more.

Reply:
spawning shrikes should be done by hive datum code not gamemode, same for queen.

Reply:
That will require a very big refactor of the game mode, which is definitely out of scope here.

Reply:
no it wouldn't.

Reply:
You'll need to suggest ways then.
Spawn all the regular xenos first, which are then added to the hive, then create a normal xeno hive variable, and trigger a proc with the xeno queen candidate, for it to sort it into a queen or shrike?

Reply:
yeah i suppose to do it properly you'd need to refactor gamemodes slightly